aiven/internal/service/static_ips.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2017 jelmersnoeck
+// Copyright (c) 2018-2021 Aiven, Helsinki, Finland. https://aiven.io/
+package service
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aiven/aiven-go-client"
+	"github.com/aiven/terraform-provider-aiven/aiven/internal/schemautil"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	staticIpAssigned  = "assigned"
+	staticIpAvailable = "available"
+)
+
+func CurrentlyAllocatedStaticIps(ctx context.Context, projectName, serviceName string, m interface{}) ([]string, error) {
+	client := m.(*aiven.Client)
+
+	// special handling for static ips
+	staticIpListResponse, err := client.StaticIPs.List(projectName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to list static ips for project '%s': %w", projectName, err)
+	}
+	allocatedStaticIps := make([]string, 0)
+	for i := range staticIpListResponse.StaticIPs {
```

Comment:
we are searching for static IP's in a list in 3 or more places, would be nice to a separate function which does it, maybe in go-client, WDYT?

Reply:
We search slightly different each time ( all ips for a service, ip with a service id, ips in several states of assignment ). I would keep it like it is for now.